### PR TITLE
feat: yaml config options for html generation using liquid templates

### PR DESCRIPTION
**Pull request by estherixz**
## Description

This is a first draft for the potential implementation of issue #24 on which I'd like to get some feedback.
The solution is very simple (maybe too simple?). I added the option to have some YAML metadata at the top of the generated readme, which can assist us in the usage of a liquid template.

This change is accompanied by changes to the gigsboat/template [PR](https://github.com/gigsboat/gigs-eleventy-template/pull/3/files) in order to give some defaults for an eleventy starter. Maybe it would be even better if we created a new _gigs-eleventy-template_ but for now the PR is based on the original. More details for the template changes will be in the relevant PR.

I tried to be as consisted as possible with the current implementation. Atm, this approach doesn't give full flexibility on the html produced, but I didn't want to deviate too much from the current. Hence, the contents of the document are not editable. If a JSON output is eventually implemented, this approach could change to give more flexible options.

Alternatively, I can use the `pages` files as the source of truth and create completely new methods for the html (or liquid template, or other eleventy compatible) file creation but I decided to go with the simplest option to start with, in order to not add too much to the existing code.

This simple approach also gives flexibility to use any html generator, not just eleventy. (ie: it is compatible with jekyll)

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Related Issue
#24 

## Motivation and Context
We can generate html in order to display gigs on a different format

## How Has This Been Tested?
Locally on macOS and with a [github pages deployment](https://estherixz.github.io/gigs-template/)
Existing tests are passing as expected and new tests have been added for the metadata config. There is an action that runs on the new template which demonstrates the steps to generate an html using eleventy.

## Screenshots (if appropriate):
Screenshot 2022-01-25 at 12 43 39

## Checklist:
- [x] I have updated the documentation (if required).
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I added a picture of ~a cute~ 11 grumpy animal(s) cause it's fun

11 cats

### CODE REVIEW

**Comment by lirantal** [comment]
Thanks for all the effort put into this @estherixz ❤️ 

If I understand correctly, the yaml frontmatter added to the README file will be later used by other static site generators to create an HTML version out of that markdown file?

By the way, are you sure the generated README.md file doesn't display the front matter part when viewed on GitHub or other markdown readers?

File: __tests__/__snapshots__/main.test.js.snap [hunk 1]
```diff
@@ -82,6 +82,109 @@ Liran Tal
...
+<i>Updated on 2021-12-14T00:00:00.000Z</i>"
+`;
+
+exports[`given a directory of markdown files a full document is rendered with any yaml metadata passed 1`] = `
+"---
+title: a title
+description: decsription of document
+randomProp: a random prop description
+randomPropNumber: 5
+---
```
Comment (lirantal) [inline __tests__/__snapshots__/main.test.js.snap:99]: This front matter part added to the README.md generated file is a requirement for those static site generators to create an HTML version of it then?

Comment (estherixz) [reply __tests__/__snapshots__/main.test.js.snap:99]: No it's not needed after all (🥳) which is why I closed this PR.

And to answer you previous question, it was being displayed in the generated README file, which was another reason why it wasn't suitable.

But I found a way to skip this part completely with `11ty` and it's definitely not needed for jekyll.

I'm just in the process of updating the documenting it.

Comment (lirantal) [reply __tests__/__snapshots__/main.test.js.snap:99]: Ok cool, I'll wait till you're finish with the changes you want to propose and we can follow up on that with more conversation :)

**Pull request closed without merging**